service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/SendDocumentTaskExecutorTest.java

```diff
@@ -1,25 +1,145 @@
+    private Gp2gpConfiguration gp2gpConfiguration;
+
     @ParameterizedTest
     @MethodSource("chunkTestData")
     void When_ChunkingString_Expect_StringIsProperlySplit(String input, int sizeThreshold, List<String> output) {
         var result = SendDocumentTaskExecutor.chunkBinary(input, sizeThreshold);
         assertThat(result).containsExactlyElementsOf(output);
     }
 
+    @BeforeEach
+    public void setUp() {
+        when(mhsClient.sendMessageToMHS(any())).thenReturn("THIS IS A TEST STRING");
+
+        this.storageConnector = createLocalStorageConnector();
+        gp2gpConfiguration = new Gp2gpConfiguration();
+        this.sendDocumentTaskExecutor = new SendDocumentTaskExecutor(
+                new StorageConnectorService(storageConnector, new ObjectMapper()),
+                this.mhsRequestBuilder,
+                this.mhsClient,
+                new RandomIdGeneratorServiceStub(),
+                this.ehrExtractStatusService,
+                new ObjectMapper(),
+                this.detectDocumentsSentService,
+                gp2gpConfiguration,
+                new EhrDocumentMapper(new TimestampService(), new RandomIdGeneratorServiceStub()),
+                new TimestampService()
+        );
+    }
+
+    @Nullable
+    private static StorageConnector createLocalStorageConnector() {
+        StorageConnectorFactory storageConnectorFactory = new StorageConnectorFactory();
+        StorageConnectorConfiguration configuration = new StorageConnectorConfiguration();
+        configuration.setType(StorageConnectorOptions.LOCALMOCK.getStringValue());
+        storageConnectorFactory.setConfiguration(configuration);
+        return storageConnectorFactory.getObject();
+    }
+
+    @Test
+    public void When_DocumentNeedsToBeSplitInto5Chunks_Expect_FiveChunksWhereMultipartContentTypeHeaderIsOctetStream() {
+        final int SIZE_OF_EACH_CHUNK = 1;
+        final int NUMBER_OF_CHUNKS = 5;
+        this.gp2gpConfiguration.setLargeAttachmentThreshold(SIZE_OF_EACH_CHUNK);
+
+        byte[] storageDataWrapper = generateStorageDataWrapper(SIZE_OF_EACH_CHUNK * NUMBER_OF_CHUNKS);
+        this.storageConnector.uploadToStorage(
+                new ByteArrayInputStream(storageDataWrapper),
+                storageDataWrapper.length,
+                "large_file_which_will_be_split.txt"
+        );
+
+        this.sendDocumentTaskExecutor.execute(
+                SendDocumentTaskDefinition.builder()
+                        .documentName("large_file_which_will_be_split.txt")
+                        .messageId("88")
+                        .fromOdsCode("RANDOM-ODS")
+                        .conversationId("RANDOM-ID")
+                        .build()
+        );
+
+        verify(mhsRequestBuilder, times(NUMBER_OF_CHUNKS)).buildSendEhrExtractCommonRequest(
+                contains("\"content_type\":\"application/octet-stream\""),
+                eq("RANDOM-ID"),
+                eq("RANDOM-ODS"),
+                any()
```

Comment:
I think slightly narrowing down the last param to anyString() is better than pass any()

Reply:
So what about this point above?